File: redis-examples/src/main/rb/io/vertx/examples/redis/redis_client_verticle.rb
require 'vertx-redis/redis_client'
# If a config file is set, read the host and port.
host = Java::IoVertxGroovyCore::Vertx.current_context().config()["host"]
if (host == nil)
  host = "127.0.0.1"
end

# Create the redis client
client = VertxRedis::RedisClient.create($vertx, {
  'host' => host
})

client.set("key", "value") { |r_err,r|
  if (r_err == nil)
    puts "key stored"
    client.get("key") { |s_err,s|
      puts "Retrieved value :#{s}"
    }
  else
    puts "Connection or Operation Failed #{r_err}"
  end
}

File: redis-examples/src/main/js/io/vertx/examples/redis/redis_client_verticle.js
var Vertx = require("vertx-js/vertx");
var RedisClient = require("vertx-redis-js/redis_client");
// If a config file is set, read the host and port.
var host = Vertx.currentContext().config().host;
if ((host === null || host === undefined)) {
  host = "127.0.0.1";
}

// Create the redis client
var client = RedisClient.create(vertx, {
  "host" : host
});

client.set("key", "value", function (r, r_err) {
  if (r_err == null) {
    console.log("key stored");
    client.get("key", function (s, s_err) {
      console.log("Retrieved value: " + s);
    });
  } else {
    console.log("Connection or Operation Failed " + r_err);
  }
});
